mne/annotations.py

```diff
@@ -443,3 +443,80 @@ def _ensure_annotation_object(obj):
     if not isinstance(obj, Annotations):
         raise ValueError('Annotations must be an instance of '
                          'mne.Annotations. Got %s.' % obj)
+
+
+@verbose
+def events_from_annotations(raw, event_id=None, regexp=None,
+                            event_id_func=None, verbose=None):
+    """Get events and event_id from an Annotations object
+
+    Parameters
+    ----------
+    raw : instance of Raw
+        The raw data for which Annotations are defined.
+    event_id : dict | None
+        Dictionary of string keys and integer values as used in mne.Epochs
+        to map annotation descriptions to integer event codes. If None,
+        all descriptions of annotations are mapped and assigned arbitrary
+        unique integer values. Else, only the keys present will be mapped
+        and and the annotations with other descriptions will be ignored.
+    regexp : str | None
+        Regular expression used to filter the annotations whose
+        descriptions is a match.
+    event_id_func : None | str | callable
+        What to do for events not found in ``event_id``. Must take one ``str``
+        argument and return an ``int``. If string, must be 'strip-to-integer',
+        in which case it defaults to stripping event codes such as "D128" or
+        "S  1" of their non-integer parts and returns the integer.
+        If the event is not in the ``event_id`` and calling ``event_id_func``
+        on it results in a ``TypeError`` (e.g. if ``event_id_func`` is
+        ``None``) or a ``ValueError``, the event is dropped.
+    verbose : bool, str, int, or None
+        If not None, override default verbose level (see
+        :func:`mne.verbose` and :ref:`Logging documentation <tut_logging>`
+        for more). Defaults to self.verbose.
+
+    Returns
+    -------
+    events : ndarray, shape (n_events, 3)
+        The events.
+    event_id : dict
+        The event_id variable that can be passed to Epochs.
+    """
+    import re
+
+    if raw.annotations is None:
```

Comment:
Could annotations be empty now? For example after adding and removing or adding and croping?

Reply:
yes,https://github.com/mne-tools/mne-python/pull/5416